### ISSUE TITLE
e2e: SchedulerPredicates validates that NodeSelector is respected

### ISSUE THREAD

**Issue by sttts** [mesosphere/kubernetes-mesos#438]
```
SchedulerPredicates
  validates that NodeSelector is respected.
  /Users/sts/Quellen/mesos/kubernetes/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:358
[BeforeEach] SchedulerPredicates
  /Users/sts/Quellen/mesos/kubernetes/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:152
>>> testContext.KubeConfig: /Users/sts/.kube/config
Aug 18 13:07:44.024: INFO: Waiting for terminating namespaces to be deleted...
Aug 18 13:07:49.040: INFO: Waiting up to 2m0s for service account default to be provisioned in ns e2e-tests-sched-pred-xz2qv
Aug 18 13:07:49.045: INFO: Service account default in ns e2e-tests-sched-pred-xz2qv had 0 secrets, ignoring for 5s: <nil>
Aug 18 13:07:54.051: INFO: Service account default in ns e2e-tests-sched-pred-xz2qv with secrets found. (5.010053416s)
[It] validates that NodeSelector is respected.
  /Users/sts/Quellen/mesos/kubernetes/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:358
STEP: Trying to schedule Pod with nonempty NodeSelector.
Aug 18 13:07:54.065: INFO: Sleeping 10 seconds and crossing our fingers that scheduler will run in that time.
[AfterEach] SchedulerPredicates
  /Users/sts/Quellen/mesos/kubernetes/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:166
STEP: Destroying namespace for this suite e2e-tests-sched-pred-xz2qv

• Failure [20.095 seconds]
SchedulerPredicates
/Users/sts/Quellen/mesos/kubernetes/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:359
  validates that NodeSelector is respected. [It]
  /Users/sts/Quellen/mesos/kubernetes/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:358


  Expected
      <[]api.Event | len:0, cap:0>: []
  not to be empty

  /Users/sts/Quellen/mesos/kubernetes/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:124
```

**Comment by sttts** [comment]
Do we support the NodeSelector in the pod spec at all in the scheduler right now?

**Comment by jdef** [comment]
We don't leverage any of the upstream scheduling predicates.

**Comment by sttts** [comment]
Any reason not to add NodeSelector support? I guess we need a richer cloud provider API to actually add labels to nodes automatically. But the e2e test here only needs very thin support for label filtering in the scheduler.

**Comment by sttts** [comment]
In fact an offer includes the slave attributes: https://github.com/apache/mesos/blob/master/include/mesos/mesos.proto#L836

**Comment by sttts** [comment]
Fixed by https://github.com/kubernetes/kubernetes/pull/12943

**Comment by sttts** [comment]
Fixed by kubernetes/kubernetes#12943
